streams/src/main/java/org/apache/kafka/streams/state/internals/RocksDBWindowStore.java

```diff
@@ -48,11 +49,12 @@
 
     private static final long USE_CURRENT_TIMESTAMP = -1L;
 
-    private static class Segment extends RocksDBStore<byte[], byte[]> {
+    // use the ByteBuffer wrapper for underlying rocksDB keys since they are used for hashing data structures
+    private static class Segment extends RocksDBStore<ByteBuffer, byte[]> {
```

Comment:
Using `ByteBuffer` works, but it's a bit dangerous because of how it's mutated (and the hashCode changed) every time a `get*` method is called. I think it would be nicer if we had an immutable class wrapping a byte array that implemented `hashCode`, `equals` and `toStrng`.